batch/batch/driver/instance_collection/base.py

```diff
@@ -87,6 +96,15 @@ def get_instance(self, inst_name) -> Optional[Instance]:
             return inst_coll.name_instance.get(inst_name)
         return None
 
+    async def get_token_from_instance_name(self, name):
+        record: Dict[str, Any] = await self.db.select_and_fetchone(
+            'SELECT token FROM instances WHERE name = %s',
+            (name),
+        )
+        if not record or 'token' not in record:
```

Comment:
If the record exists is not None, then you should be guaranteed to have the token field in it.